test/hotspot/jtreg/compiler/escapeAnalysis/Test8329757.java

```diff
@@ -43,7 +43,7 @@ public class Test8329757 {
 
     void q() {
         int e;
-        synchronized (new Double(1.1f)) {
+        synchronized (new Object()) {
```

Comment:
What motivated this change?

Reply:
We cannot synchronize on value classes (primitive boxes are migrated to value classes with preview mode enabled) and we throw an `IdentityException` if we attempt to do so. I've talked with the author of the test and compiler folks and it doesn't change the semantics if we change the mutex.

Reply:
I think it makes sense to push this and similar changes into jdk/jdk to minimize diff for valhalla.